libraries/authentication.rb

```diff
@@ -1,134 +1,107 @@
+        'cacheTiming' => 'Scripted',
+        'scriptPath' => 'Scripted',
+        'scriptSearchFilters' => 'Scripted',
+        'passwordHashAlgorithm' => 'Splunk'
+      }.freeze
+
+      probable_types = config_mapping.values_at(*settings.keys).uniq
+      raise "Conflicting authentication types were derived from the config: #{probable_types.join(',')}" if probable_types.length > 1
+
+      settings['authType'] || probable_types.first || 'Splunk'
+    end
+
     def self.configure_authentication(node, hash) # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity, MethodLength
       hash = hash.clone
       auth_stanzas = { 'authentication' => hash }
-      fail 'authSettings is managed by chef. Don\'t set it yourself!' if hash.key?('authSettings')
-
-      unless hash['authType']
-        guesses = ASSUMPTIONS.inject([]) do |result, (key, type)|
-          result << type if hash.key?(key) && !result.include?(type)
-          result
-        end
-
-        hash['authType'] =
-          case guesses.length
-          when 0
-            'Splunk'
-          when 1
-            guesses.first
-          else
-            fail "Unable to determine authType! Guesses include #{guesses.join(',')}"
-          end
-      end
+      raise "authSettings is managed by Chef. Don't set it yourself!" if hash.key?('authSettings')
 
-      ASSUMPTIONS.each do |key, type|
-        if hash['authType'] != type && hash.key?(key)
-          fail "#{key} is only supported with #{type}. authType = #{hash['authType']}"
-        end
-      end
+      hash['authType'] = determine_auth_type(hash)
 
       default_coords = CernerSplunk::DataBag.to_a node['splunk']['config']['authentication']
 
       case hash['authType']
-      when 'Splunk'
-      # Nothing special to do here.
+      when 'Splunk' then nil # Nothing special to do here.
       when 'Scripted'
-        script = {
-          'scriptPath' => hash.delete('scriptPath')
-        }
-        fail 'scriptPath required for Scripted authentication' unless script['scriptPath']
-        search_filters = hash.delete('scriptSearchFilters')
-        script['scriptSearchFilters'] = search_filters if search_filters
-        hash['authSettings'] = 'script'
+        raise 'scriptPath required for Scripted authentication' unless hash['scriptPath']
+        script = { 'scriptPath' => hash.delete('scriptPath') }
+        script['scriptSearchFilters'] = hash.delete 'scriptSearchFilters' if hash.key? 'scriptSearchFilters'
         auth_stanzas['script'] = script
-        cache_timing = hash.delete('cacheTiming')
-        if cache_timing
-          fail "Unknown type for cacheTiming: #{cacheTiming.class}" unless cache_timing.is_a?(Hash)
-          auth_stanzas['cacheTiming'] = cache_timing
+        hash['authSettings'] = 'script'
+
+        if hash.key? 'cacheTiming'
+          raise "Unknown type for cacheTiming: #{cacheTiming.class}" unless hash['cacheTiming'].is_a?(Hash)
```

Comment:
The variable cacheTiming from the message doesn't exist.  (it's an existing issue obviously)